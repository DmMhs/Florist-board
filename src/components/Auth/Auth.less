.Auth {
  min-height: calc(100vh - 7rem);
  form {
    display: flex;
    flex-direction: column;
    justify-content: center;
    align-items: center;
    width: 50%;
    margin: 0 auto;
    border: 1px solid #ccc;
    border-radius: 8px;
    padding: 1rem;
    .authenticated-message {
      color: #333333;
      h2 {
        font-weight: normal;
        i {
          font-size: 2.5rem;
        }
      }
      span {
        font-size: 1.3rem;
      }
    }
    .form-field {
      width: calc(100% - 2rem);
      margin: 0 1rem;
      font-size: 1.3rem;
      display: flex;
      flex-direction: row;
      justify-content: flex-start;
      flex-wrap: wrap;
      align-items: center;
      padding: 0.7rem;
      label {
        width: 100%;
        text-align: left;
      }
      input {
        width: 50%;
        font-size: 1.2rem;
        padding: 5px;
        border-radius: 5px;
        border: 1px solid #ccc;
      }
    }
    button {
      border: none;
      border-radius: 5px;
      padding: 7px;
      background-color: rgb(98, 182, 103);
      font-size: 1.3rem;
      font-weight: lighter;
      color: white;
      cursor: pointer;
      &:hover {
        background-color: rgb(78, 145, 83);
      }
      &[disabled] {
        background-color: rgb(185, 212, 185);
        pointer-events: none;
      }
    }
    .Popup {
      margin: 0 8px;
      padding: 5px;
      i {
        margin: 0;
        margin-right: 5px;
      }
      i,
      span {
        font-size: 1.1rem;
      }
    }
    hr {
      border: 0.5px solid #eee;
      width: 90%;
    }
    h3 {
      font-weight: normal;
      color: #333333;
    }
    p {
      width: 100%;
      text-align: left;
    }
    .switch-btn {
      background-color: skyblue;
      &:hover {
        background-color: rgb(105, 163, 185);
      }
      a {
        text-decoration: none;
        color: white;
      }
    }
    .google-pic,
    .facebook-pic {
      max-width: 2rem;
      margin: 0 10px;
      cursor: pointer;
      transition: 1s;
      &:hover {
        position: relative;
        bottom: 1px;
        left: 1px;
        transition: 1s;
      }
    }
  }
}

@media screen and (max-width: 768px) {
  .Auth {
    width: 100%;
    form {
      width: 80%;
      margin-top: 4rem;
      margin-bottom: 1rem;
      .form-field {
        font-size: 1.1rem;
        input {
          width: 95%;
        }
      }
    }
    button {
      font-size: 1.1rem;
      margin: 0 auto;
      margin-bottom: 5px;
    }
    .Popup {
      padding: 0px;
      width: fit-content;
    }
  }
}
