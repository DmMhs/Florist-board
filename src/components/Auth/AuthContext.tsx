import React, { Component } from 'react';
import { withRouter, RouteComponentProps } from 'react-router-dom';

export interface AuthContextState {
  userLogin: string | null | undefined;
  userId: string | null | undefined;
  userToken: string | null | undefined;
  userAuthenticated: boolean;
  authenticationMethod: string | null | undefined;
  lang: string | null | undefined;
}

export const AuthContext = React.createContext({
  state: {} as AuthContextState,
  setUserCredentials: (
    userLogin: string | null,
    userId: string | null,
    userToken: string | null,
    authenticationMethod?: string | null,
    event?: Event
  ) => {},
  logout: () => {},
  setLang: (lang: string | null) => {}
});
class AuthContextProvider extends Component<
  RouteComponentProps<{}>,
  AuthContextState
> {
  constructor(props: RouteComponentProps<{}>) {
    super(props);
    this.state = {
      userLogin: '',
      userId: '',
      userToken: '',
      userAuthenticated: false,
      authenticationMethod: undefined,
      lang: 'en'
    };
  }
  componentDidMount() {
    if (
      localStorage.floristAuthToken === '' ||
      localStorage.floristAuthToken === undefined
    ) {
      this.setState({
        userLogin: '',
        userId: '',
        userToken: '',
        authenticationMethod: '',
        userAuthenticated: false
      });
    } else {
      this.setState({
        userLogin: localStorage.floristAuthLogin,
        userId: localStorage.floristAuthUserId,
        userToken: localStorage.floristAuthToken,
        authenticationMethod: localStorage.floristAuthMethod,
        userAuthenticated: true
      });
    }
  }
  setUserCredentialsHandler = (
    userLogin: string | null | undefined,
    userId: string | null | undefined,
    userToken: string | null | undefined,
    authenticationMethod?: string | null | undefined,
    event?: Event
  ) => {
    this.setState({
      userAuthenticated: true,
      userLogin,
      userId,
      userToken,
      authenticationMethod
    });
    localStorage.floristAuthLogin = userLogin;
    localStorage.floristAuthToken = userToken;
    localStorage.floristAuthUserId = userId;
    localStorage.floristAuthMethod = authenticationMethod;
  };
  logoutHandler = () => {
    this.setState({
      userAuthenticated: false,
      userLogin: '',
      userId: '',
      userToken: '',
      authenticationMethod: undefined
    });
    localStorage.floristAuthLogin = '';
    localStorage.floristAuthToken = '';
    localStorage.floristAuthUserId = '';
    localStorage.floristAuthMethod = '';
  };

  setLangHandler = (lang: string | null) => {
    this.setState({
      lang: lang
    });
  };

  render() {
    return (
      <AuthContext.Provider
        value={{
          state: this.state,
          setUserCredentials: this.setUserCredentialsHandler,
          logout: this.logoutHandler,
          setLang: this.setLangHandler
        }}
      >
        {this.props.children}
      </AuthContext.Provider>
    );
  }
}

export default withRouter<RouteComponentProps<{}>>(AuthContextProvider);
