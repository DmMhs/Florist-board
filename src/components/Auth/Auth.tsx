import React, { Component } from 'react';
import firebase from 'firebase';
import axios from 'axios';
import {
  withRouter,
  RouteComponentProps as RCProps,
  NavLink
} from 'react-router-dom';

import Popup from '../Popup/Popup';
import { AppContext } from '../../AppContext';
import labels from '../../config/labels';
import './Auth.less';

interface MatchParams {
  mode: string;
}

interface Props extends RouteComponentProps<MatchParams> {}

interface RouteComponentProps<P> {
  match: match<P>;
}

interface match<P> {
  params: P;
  isExact: boolean;
  path: string;
  url: string;
}

interface AuthState {
  formData: {
    email: string;
    password: string;
  };
  mode: string;
}

class Auth extends Component<
  RouteComponentProps<MatchParams> & RCProps<{}>,
  AuthState
> {
  public static getDerivedStateFromProps(
    props: RouteComponentProps<MatchParams> & RCProps<{}>,
    state: AuthState
  ) {
    return {
      mode: props.match.params.mode
    };
  }

  constructor(props: RouteComponentProps<MatchParams> & RCProps<{}>) {
    super(props);
    this.state = {
      formData: {
        email: '',
        password: ''
      },
      mode: this.props.match.params.mode
    };
  }

  private emailInputChangeHandler = (
    event: React.ChangeEvent<HTMLInputElement>
  ) => {
    const updatedFormData = { ...this.state.formData };
    updatedFormData.email = event.target.value;
    this.setState({
      formData: updatedFormData
    });
  };

  private passwordInputChangeHandler = (
    event: React.ChangeEvent<HTMLInputElement>
  ) => {
    const updatedFormData = { ...this.state.formData };
    updatedFormData.password = event.target.value;
    this.setState({
      formData: updatedFormData
    });
  };

  private getIdToken = () => {
    return firebase
      .auth()
      .currentUser!.getIdToken(true)
      .then(idToken => {
        return idToken;
      })
      .catch(error => {
        console.log(error);
      });
  };

  private formSubmitHandler = (
    event: React.MouseEvent<HTMLFormElement, MouseEvent>
  ) => {
    event.preventDefault();
    const value = this.context;
    const initialFormData = {
      email: '',
      password: ''
    };
    const authData = {
      email: this.state.formData.email,
      password: this.state.formData.password,
      returnSecureToken: true
    };

    if (this.state.mode === 'signup') {
      this.setState({
        formData: initialFormData
      });
      firebase
        .auth()
        .createUserWithEmailAndPassword(authData.email, authData.password)
        .then(response => {
          (this.props as RouteComponentProps<MatchParams> &
            RCProps<{}>).history.push('/auth/signin');
        })
        .catch(error => {
          console.log(error);
        });
    } else {
      this.setState({
        formData: initialFormData
      });
      firebase
        .auth()
        .signInWithEmailAndPassword(authData.email, authData.password)
        .then(async response => {
          const idToken = await this.getIdToken();
          value.setUserCredentials(
            response.user!.email,
            response.user!.uid,
            idToken
          );
          (this.props as RouteComponentProps<MatchParams> &
            RCProps<{}>).history.push('/');
        })
        .catch(error => console.log(error));
    }
  };

  private authWithGoogleHandler = () => {
    const value = this.context;
    firebase
      .auth()
      .signInWithPopup(new firebase.auth.GoogleAuthProvider())
      .then(async response => {
        const idToken = await this.getIdToken();
        value.setUserCredentials(
          response.user!.email,
          response.user!.uid,
          idToken,
          'google'
        );
        await axios.post(
          'https://us-central1-florist-cb933.cloudfunctions.net/giveAdminRole',
          {
            admin: true
          }
        );
        (this.props as RouteComponentProps<MatchParams> &
          RCProps<{}>).history.push('/');
      })
      .catch(error => console.log(error));
  };

  private authWithFacebookHandler = () => {
    const value = this.context;
    firebase
      .auth()
      .signInWithPopup(new firebase.auth.FacebookAuthProvider())
      .then(async response => {
        const idToken = await this.getIdToken();
        value.setUserCredentials(
          response.user!.displayName,
          response.user!.uid,
          idToken,
          'facebook'
        );
        (this.props as RouteComponentProps<MatchParams> &
          RCProps<{}>).history.push('/');
      })
      .catch(error => console.log(error));
  };

  public render() {
    const { email, password } = this.state.formData;
    const { mode } = this.state;
    const context = this.context;

    const infoPopup = (
      <Popup
        type="info"
        message={
          context.state.lang === 'en'
            ? 'Please, provide a required data'
            : 'Будь ласка, введіть дані'
        }
      />
    );
    return (
      <div className="Auth">
        <AppContext.Consumer>
          {value =>
            value && (
              <form
                onSubmit={
                  this.formSubmitHandler as
                    | ((event: React.FormEvent<HTMLFormElement>) => void)
                    | undefined
                }
              >
                {value.state.userAuthenticated === false ? (
                  <div>
                    <div className="form-field">
                      <label>
                        {labels[value.state.lang as string].pages.auth.email}
                      </label>
                      <input
                        type="email"
                        className="email-input"
                        onChange={this.emailInputChangeHandler}
                        value={email}
                        required
                      />
                    </div>
                    <div className="form-field">
                      <label>
                        {labels[value.state.lang as string].pages.auth.password}
                      </label>
                      <input
                        type="password"
                        className="password-input"
                        minLength={6}
                        onChange={this.passwordInputChangeHandler}
                        value={password}
                        required
                      />
                    </div>
                    <div className="form-field">
                      <button
                        type="submit"
                        disabled={
                          password === '' || email === '' ? true : false
                        }
                        className="submit-btn"
                      >
                        {labels[value.state.lang as string].pages.auth.btn}
                      </button>
                      {password === '' || email === '' ? infoPopup : null}
                    </div>
                    <hr />
                    <h3>
                      {
                        labels[value.state.lang as string].pages.auth
                          .alternative
                      }
                    </h3>
                    <div className="form-field">
                      <p>
                        {mode === 'signup'
                          ? labels[value.state.lang as string].pages.auth.signup
                              .google
                          : labels[value.state.lang as string].pages.auth.signin
                              .google}
                      </p>{' '}
                      <br />
                      <a
                        onClick={this.authWithGoogleHandler}
                        className="google-auth"
                      >
                        <img
                          src="https://firebasestorage.googleapis.com/v0/b/florist-cb933.appspot.com/o/icons%2Fgoogle.png?alt=media&token=3210d61d-cad2-45bc-a343-0ed08c097bb6"
                          alt="google-pic"
                          className="google-pic"
                        />
                      </a>
                    </div>
                    <div className="form-field">
                      <p>
                        {mode === 'signup'
                          ? labels[value.state.lang as string].pages.auth.signup
                              .facebook
                          : labels[value.state.lang as string].pages.auth.signin
                              .facebook}
                      </p>
                      <br />
                      <a
                        onClick={this.authWithFacebookHandler}
                        className="facebook-auth"
                      >
                        <img
                          src="https://firebasestorage.googleapis.com/v0/b/florist-cb933.appspot.com/o/icons%2Ffacebook.png?alt=media&token=32a65be5-5336-45fd-8a50-44f8880ddbd0"
                          alt="facebook-pic"
                          className="facebook-pic"
                        />
                      </a>
                    </div>
                    <hr />
                    <div className="form-field">
                      <p>
                        {mode === 'signup'
                          ? labels[value.state.lang as string].pages.auth.signup
                              .account
                          : labels[value.state.lang as string].pages.auth.signin
                              .account}
                      </p>
                      <button type="button" className="switch-btn">
                        <NavLink
                          to={`/auth/${
                            mode === 'signup' ? 'signin' : 'signup'
                          }`}
                        >
                          {mode === 'signup'
                            ? labels[value.state.lang as string].pages.auth
                                .signup.btn
                            : labels[value.state.lang as string].pages.auth
                                .signin.btn}
                        </NavLink>
                      </button>
                    </div>
                  </div>
                ) : (
                  <div className="authenticated-message">
                    {' '}
                    <h2>
                      {
                        labels[value.state.lang as string].pages.auth.isAuth
                          .main
                      }{' '}
                      <i className="far fa-flushed" />
                    </h2>{' '}
                    <span>
                      {labels[value.state.lang as string].pages.auth.isAuth.sub}
                    </span>
                  </div>
                )}
              </form>
            )
          }
        </AppContext.Consumer>
      </div>
    );
  }
}

Auth.contextType = AppContext;
export default withRouter<RouteComponentProps<MatchParams> & RCProps<{}>>(Auth);
