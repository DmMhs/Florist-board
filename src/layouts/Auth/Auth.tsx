import React, { Component } from 'react';
import firebase from 'firebase';
import axios from 'axios';
import {
  withRouter,
  RouteComponentProps as RCProps,
  NavLink
} from 'react-router-dom';

import { Popup } from '../../components';
import { AppContext } from '../../AppContext';
import { userRole } from '../../services/auth/userRole';
import { getIdToken } from '../../services/auth/getIdToken';
import { createUserWithEmailAndPassword } from '../../services/auth/createUserWithEmailAndPassword';
import { signInWithEmailAndPassword } from '../../services/auth/signInWithEmailAndPassword';
import './Auth.less';

interface MatchParams {
  mode: string;
}

interface Props extends RouteComponentProps<MatchParams> {}

interface RouteComponentProps<P> {
  match: match<P>;
}

interface match<P> {
  params: P;
  isExact: boolean;
  path: string;
  url: string;
}

interface AuthState {
  formData: {
    email: string;
    password: string;
  };
  mode: string;
}

class Auth extends Component<
  RouteComponentProps<MatchParams> & RCProps<{}>,
  AuthState
> {
  public static getDerivedStateFromProps(
    props: RouteComponentProps<MatchParams> & RCProps<{}>,
    state: AuthState
  ) {
    return {
      mode: props.match.params.mode
    };
  }

  constructor(props: RouteComponentProps<MatchParams> & RCProps<{}>) {
    super(props);
    this.state = {
      formData: {
        email: '',
        password: ''
      },
      mode: this.props.match.params.mode
    };
  }

  private redirectHandler = (path: string) => {
    (this.props as RouteComponentProps<MatchParams> & RCProps<{}>).history.push(
      path
    );
  };

  private emailInputChangeHandler = (
    event: React.ChangeEvent<HTMLInputElement>
  ) => {
    const updatedFormData = { ...this.state.formData };
    updatedFormData.email = event.target.value;
    this.setState({
      formData: updatedFormData
    });
  };

  private passwordInputChangeHandler = (
    event: React.ChangeEvent<HTMLInputElement>
  ) => {
    const updatedFormData = { ...this.state.formData };
    updatedFormData.password = event.target.value;
    this.setState({
      formData: updatedFormData
    });
  };

  private formSubmitHandler = async (
    event: React.FormEvent<HTMLFormElement>
  ) => {
    event.preventDefault();
    const value = this.context;
    const initialFormData = {
      email: '',
      password: ''
    };
    const authData = {
      email: this.state.formData.email,
      password: this.state.formData.password,
      returnSecureToken: true
    };

    if (this.state.mode === 'signup') {
      this.setState({
        formData: initialFormData
      });
      await createUserWithEmailAndPassword(authData.email, authData.password);
      this.redirectHandler('/auth/signin');
    } else {
      this.setState({
        formData: initialFormData
      });
      await signInWithEmailAndPassword(authData.email, authData.password)
        .then(async response => {
          const idToken = await getIdToken();
          value.setUserCredentials(
            response.user!.email,
            response.user!.uid,
            idToken
          );
        })
        .catch(err => console.log(err));

      this.redirectHandler('/');
    }
  };

  private authWithGoogleHandler = () => {
    const value = this.context;
    firebase
      .auth()
      .signInWithPopup(new firebase.auth.GoogleAuthProvider())
      .then(async response => {
        axios
          .post(
            'https://us-central1-florist-cb933.cloudfunctions.net/assignUserRole',
            null,
            {
              params: {
                id: response.user!.uid
              }
            }
          )
          .catch(err => console.log(err));

        const idToken = await getIdToken();
        const role = await userRole();

        value.setUserCredentials(
          response.user!.email,
          response.user!.uid,
          idToken,
          'google',
          role
        );

        this.redirectHandler('/');
      })
      .catch(error => console.log(error));
  };

  private authWithFacebookHandler = () => {
    const value = this.context;
    firebase
      .auth()
      .signInWithPopup(new firebase.auth.FacebookAuthProvider())
      .then(async response => {
        axios.post(
          'https://us-central1-florist-cb933.cloudfunctions.net/assignUserRole',
          null,
          {
            params: {
              id: response.user!.uid
            }
          }
        );

        const idToken = await getIdToken();
        const role = await userRole();

        value.setUserCredentials(
          response.user!.displayName,
          response.user!.uid,
          idToken,
          'facebook',
          role
        );

        this.redirectHandler('/');
      })
      .catch(error => console.log(error));
  };

  public render() {
    const { email, password } = this.state.formData;
    const { mode } = this.state;
    const context = this.context;
    const lang = context.state.lang;
    const labels = context.state.labels;

    const infoPopup = (
      <Popup
        type="info"
        message={
          lang === 'en'
            ? 'Please, provide a required data'
            : 'Будь ласка, введіть дані'
        }
      />
    );
    return (
      <div className="Auth">
        <AppContext.Consumer>
          {value =>
            value && (
              <form
                onSubmit={
                  this.formSubmitHandler as
                    | ((event: React.FormEvent<HTMLFormElement>) => void)
                    | undefined
                }
              >
                {value.state.userAuthenticated === false ? (
                  <div>
                    <div className="form-field">
                      <label>{labels[lang].pages.auth.email}</label>
                      <input
                        type="email"
                        className="email-input"
                        onChange={this.emailInputChangeHandler}
                        value={email}
                        required
                      />
                    </div>
                    <div className="form-field">
                      <label>{labels[lang].pages.auth.password}</label>
                      <input
                        type="password"
                        className="password-input"
                        minLength={6}
                        onChange={this.passwordInputChangeHandler}
                        value={password}
                        required
                      />
                    </div>
                    <div className="form-field">
                      <button
                        type="submit"
                        disabled={
                          password === '' || email === '' ? true : false
                        }
                        className="submit-btn"
                      >
                        {labels[lang].pages.auth.btn}
                      </button>
                      {password === '' || email === '' ? infoPopup : null}
                    </div>
                    <hr />
                    <h3>{labels[lang].pages.auth.alternative}</h3>
                    <div className="form-field google-auth-field">
                      <p>
                        {mode === 'signup'
                          ? labels[lang].pages.auth.signup.google
                          : labels[lang].pages.auth.signin.google}
                      </p>{' '}
                      <br />
                      <a
                        onClick={this.authWithGoogleHandler}
                        className="google-auth"
                      >
                        <img
                          src="https://firebasestorage.googleapis.com/v0/b/florist-cb933.appspot.com/o/icons%2Fgoogle.png?alt=media&token=3210d61d-cad2-45bc-a343-0ed08c097bb6"
                          alt="google-pic"
                          className="google-pic"
                        />
                      </a>
                    </div>
                    <div className="form-field facebook-auth-field">
                      <p>
                        {mode === 'signup'
                          ? labels[lang].pages.auth.signup.facebook
                          : labels[lang].pages.auth.signin.facebook}
                      </p>
                      <br />
                      <a
                        onClick={this.authWithFacebookHandler}
                        className="facebook-auth"
                      >
                        <img
                          src="https://firebasestorage.googleapis.com/v0/b/florist-cb933.appspot.com/o/icons%2Ffacebook.png?alt=media&token=32a65be5-5336-45fd-8a50-44f8880ddbd0"
                          alt="facebook-pic"
                          className="facebook-pic"
                        />
                      </a>
                    </div>
                    <hr />
                    <div className="form-field">
                      <p>
                        {mode === 'signup'
                          ? labels[lang].pages.auth.signup.account
                          : labels[lang].pages.auth.signin.account}
                      </p>
                      <button type="button" className="switch-btn">
                        <NavLink
                          to={`/auth/${
                            mode === 'signup' ? 'signin' : 'signup'
                          }`}
                        >
                          {mode === 'signup'
                            ? labels[lang].pages.auth.signup.btn
                            : labels[lang].pages.auth.signin.btn}
                        </NavLink>
                      </button>
                    </div>
                  </div>
                ) : (
                  <div className="authenticated-message">
                    {' '}
                    <h2>
                      {
                        labels[value.state.lang as string].pages.auth.isAuth
                          .main
                      }{' '}
                      <i className="far fa-flushed" />
                    </h2>{' '}
                    <span>
                      {labels[value.state.lang as string].pages.auth.isAuth.sub}
                    </span>
                  </div>
                )}
              </form>
            )
          }
        </AppContext.Consumer>
      </div>
    );
  }
}

Auth.contextType = AppContext;
export default withRouter<RouteComponentProps<MatchParams> & RCProps<{}>>(Auth);
