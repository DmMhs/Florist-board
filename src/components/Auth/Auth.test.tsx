import React from 'react';
import { shallow, mount } from 'enzyme';

import Auth from './Auth';
import AuthContext from './AuthContext';

it('Auth component matches a snapshot', () => {
  const wrapper = shallow(<Auth />);
  expect(wrapper).toMatchSnapshot();
});

it('AuthContext component matches a snapshot', () => {
  const wrapper = shallow(<AuthContext />);
  expect(wrapper).toMatchSnapshot();
});

it('localStorage has impact on the state', () => {
  //   const match = { params: { mode: 'signin' } };
  //   const wrapper = shallow(<Auth.WrappedComponent match={match} />);
  //   const instance = wrapper.instance();
  //   instance.setState({
  //     formData: {
  //       email: 'gdsgdsg@safasf.com',
  //       password: 'asfasfasf'
  //     }
  //   });
  //   console.log(
  //     wrapper
  //       .find(<AuthContext.Consumer />)
  //       .dive()
  //       .contains('form')
  //   );
  //   expect(instance.state.formData.email).toEqual('');
  //   console.log(instance.state);
});
