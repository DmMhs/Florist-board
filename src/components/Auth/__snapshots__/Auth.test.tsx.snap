// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Auth component matches a snapshot 1`] = `
<Route>
  <Component />
</Route>
`;

exports[`AuthContext component matches a snapshot 1`] = `
<Route>
  <Component />
</Route>
`;
